### PR TITLE
Use YaruDetailPage to limit animation rect

### PR DESCRIPTION
**Pull request by Feichtmeier**
Before:

https://user-images.githubusercontent.com/15329494/213411638-fcbad16c-57e3-44f0-9896-c0f2044cfb2c.mp4

After:

https://user-images.githubusercontent.com/15329494/213548747-b55a1eb3-de50-4ea7-bd78-de0f8f18e897.mp4

### CODE REVIEW

**Comment by Feichtmeier** [comment]
@jpnurmi as described on discord:

testing your latest fix for navigatoinpage hero
and there seems to be a difference to scaffold in terms of default tab controllers

hierarchy in updates page is

Defaulttabcontroller -> scaffold -> YaruWindowtiltebar -> container -> Tabbbar

when I exchange scaffold with the yarudetailspage here i get an error on the second time (not the first) i navigate to the updates page

```
Reloaded 1 of 1691 libraries in 1.308ms (compile: 60 ms, reload: 816 ms, reassemble: 415 ms).

════════ Exception caught by widgets library ═══════════════════════════════════
The following assertion was thrown building Container(bg: BoxDecoration(borderRadius: BorderRadius.circular(6.0)), constraints: BoxConstraints(0.0<=w<=Infinity, h=34.0)):
No TabController for TabBar.

When creating a TabBar, you must either provide an explicit TabController using the "controller" property, or you must ensure that there is a DefaultTabController above the TabBar.
In this case, there was neither an explicit controller nor a default controller.
The relevant error-causing widget was
Container
When the exception was thrown, this was the stack
#0      _TabBarState._updateTabController.<anonymous closure>
#1      _TabBarState._updateTabController
#2      _TabBarState.didChangeDependencies
#3      StatefulElement._firstBuild
#4      ComponentElement.mount
...     Normal element mounting (48 frames)
#52     Element.inflateWidget
#53     MultiChildRenderObjectElement.inflateWidget
#54     MultiChildRenderObjectElement.mount
...     Normal element mounting (213 frames)
#267    Element.inflateWidget
#268    MultiChildRenderObjectElement.inflateWidget
#269    Element.updateChild
#270    RenderObjectElement.updateChildren
#271    MultiChildRenderObjectElement.update
#272    Element.updateChild
#273    ComponentElement.performRebuild
#274    StatefulElement.performRebuild
#275    Element.rebuild
#276    BuildOwner.buildScope
#277    WidgetsBinding.drawFrame
#278    RendererBinding._handlePersistentFrameCallback
#279    SchedulerBinding._invokeFrameCallback
#280    SchedulerBinding.handleDrawFrame
#281    SchedulerBinding._handleDrawFrame
#282    _invoke (dart:ui/hooks.dart:148:13)
#283    PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:318:5)
#284    _drawFrame (dart:ui/hooks.dart:115:31)
════════════════════════════════════════════════════════════════════════════════

════════ Exception caught by widgets library ═══════════════════════════════════
'package:flutter/src/widgets/framework.dart': Failed assertion: line 5334 pos 14: '_dependents.isEmpty': is not true.
The relevant error-causing widget was
YaruNavigationPage-[<null>]
════════════════════════════════════════════════════════════════════════════════
```

**Comment by jpnurmi** [comment]
@Feichtmeier Hero animations work so that hero-tagged widgets are temporarily reparented into the Navigator's overlay during page transitions. This doesn't play well together with DefaultTabController because a hero-animated child would not be able to find the controller during page transitions because the widget is temporarily moved elsewhere in the tree...

**Comment by jpnurmi** [comment]
Using a plain TabController instead is indeed the easiest solution. :)

File: lib/app/updates/updates_page.dart [hunk 1]
```diff
@@ -42,12 +42,15 @@ class UpdatesPage extends StatefulWidget {
...
 
-class _UpdatesPageState extends State<UpdatesPage> {
+class _UpdatesPageState extends State<UpdatesPage>
+    with TickerProviderStateMixin {
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
     final packageService = getService<PackageService>();
     final padding = 0.0004 * pow((widget.windowWidth * 0.85), 2);
+    final tabController =
```
Comment (jpnurmi) [inline lib/app/updates/updates_page.dart:52]: create the controller outside the `build()` method, make it a member of `_UpdatesPageState`:
```
late final TabController _tabController;

@override
void initState() {
  super.initState();
  _tabController =
      TabController(length: 2, initialIndex: widget.tabIndex, vsync: this);
}

@override
void dispose() {
  _tabController.dispose();
  super.dispose();
}
```

Comment (Feichtmeier) [reply lib/app/updates/updates_page.dart:52]: yeah ofc! sorry was on a rush on the train :)

File: lib/app/common/message_bar.dart [hunk 2]
```diff
@@ -35,11 +35,13 @@ class MessageBar extends StatelessWidget {
             ),
             onPressed: () => Clipboard.setData(ClipboardData(text: message)),
           ),
-          trailing: YaruCloseButton(
-            onPressed: () {
-              ScaffoldMessenger.of(context).clearSnackBars();
-            },
-          ),
+          actions: [
```
Comment (jpnurmi) [inline lib/app/common/message_bar.dart:38]: future improvement idea unless you like the big close button :)
```dart
onClose: (context) => ScaffoldMessenger.of(context).clearSnackBars()
```

**Review by jpnurmi — APPROVED** [review]
LGTM :+1:

File: lib/app/common/message_bar.dart (hunk 2 above)
Comment (Feichtmeier) [reply lib/app/common/message_bar.dart]: no, and I had exactly this when I was on the other branch! THanks for reminding me. Too many things in parallel again

**Pull request merged**